src/main/java/com/vmware/vipclient/i18n/messages/service/ComponentService.java

```diff
@@ -49,8 +50,11 @@ private void refreshCacheItem(final MessageCacheItem cacheItem, Iterator<DataSou
 			return;
 		}
 
-		long timestampOld = cacheItem.getTimestamp();
 		DataSourceEnum dataSource = msgSourceQueueIter.next();
+		if (!proceed(dataSource))
+			refreshCacheItem(cacheItem, msgSourceQueueIter); // Try the next dataSource in the queue
```

Comment:
It will continue execute the following code after this line finish.
So either add return statement after this line or put it into if-else block.

Reply:
I had updated this yesterday. Perhaps you did not see?